oldabe/money_in.py

```diff
@@ -504,22 +519,26 @@ def get_amounts_owed(total_amount, attributions):
             for email, share in attributions.items()}
 
 
-def distribute_remaining_amount(remaining_amount, truncated_payable_attributions, payment):
+def redistribute_pot(redistribution_pot, attributions, unpayable_contributors, payment_file, amounts_owed):
     """
-    After paying off debts, distribute remaining amount to payable contributors
-    in the attributions file, by renormalizing after excluding unpayable
-    contributors.
+    Redistribute the pot of remaining money over all payable contributors, according to attributions
+    share (normalized to 100%). Create advances for those amounts (because they are in excess
+    of the amount owed to each contributor from the original payment) and add the amounts to the
+    amounts_owed dictionary to keep track of the full amount we are about to pay everyone.
     """
-    amounts_owed = get_amounts_owed(remaining_amount, truncated_payable_attributions)
-    remainder_attributions = normalize(truncated_payable_attributions)
-    advances
-    transactions = generate_transactions(
-        remaining_amount, remainder_attributions, payment.file, commit_hash
+    fresh_advances = []
+    normalized_payable_attributions = normalize(
+        {email: share for email, share in attributions.items() if email not in unpayable_contributors}
```

Comment:
nit: We have a lot of list and dict comprehensions. I know these are small, but it could still be easier to read if we used helper functions or methods to do these things, and they'd also be reusable in case we need to do these more than once.

Reply:
Maybe so. I do hate list/dict comprehensions and find them difficult to parse visually.

One potential drawback with putting some of these in their own functions is that you might have to jump around and look for function definitions instead of being able to read through the logic in one place.

But I think it would definitely be worth putting some of these in an actual function if we do the same comprehension in multiple places. Less potential for typos when you find yourself having to re-write the comprehension instead of just calling a function that we know should be trustworthy.

Reply:
I personally love comprehensions, specifically because I find them easy to para when inlined 😂 If we are putting them in their own functions we might as well change them to a for loop.